netdata/netdata
# Add Collector log

##### Summary
Fixes https://github.com/netdata/netdata/issues/13823
Fixes #6038 

This PR has the main goal to separate `collector logs` from `error.log` simplifying netdata debug, development, and mainly helping users to find proper issues with collectors.

Some of `streaming` logs are going to be redirect for new `collector.log`, this is because our stream works like a collector (it uses the same protocol to communicate), but I did not redirect everything with this PR, because I understand this is not the main goal.
##### Test Plan

1. Compile this branch
2. Start netdata
3. Verify that errors delivered by collectors are going to new `collector.log`.

##### Additional Information
This PR is moving logs from external and internal collectors to `collector.log`.

<details> <summary>For users: How does this change affect me?</summary>
Describe the PR affects users: 
- Which area of Netdata is affected by the change? collectors
- Can they see the change or is it an under the hood? If they can see it, where? Yes, they will have separated logs for collectors.
- How is the user impacted by the change?  Simplify search for errors.
- What are there any benefits of the change?  Users have separated logs to report error for us.
</details>
